pygrocy/grocy_api_client.py

```diff
@@ -394,6 +400,40 @@ def consume_product(
 
         self._do_post_request(f"stock/products/{product_id}/consume", data)
 
+    def add_product_by_barcode(
+        self,
+        barcode: str,
+        amount: float,
+        price: float,
+        best_before_date: datetime = None,
+        transaction_type: TransactionType = TransactionType.PURCHASE,
+    ):
+        data = {
+            "amount": amount,
+            "transaction_type": transaction_type.value,
+            "price": price,
+        }
+
+        if best_before_date is not None:
+            data["best_before_date"] = best_before_date.strftime("%Y-%m-%d")
```

Comment:
IIRC there should be a helper method for date formatting which incorporates stuff like local time.

Reply:
Done :)

Reply:
I think I meant [this](https://github.com/SebRut/pygrocy/blob/develop/pygrocy/utils.py#L42) but this method only adds timezone information etc to datetime object so the serialization would have to happen. I guess just adding `localize_datetime` would be fine

Reply:
I have added this